smartlearner/interfaces/loss.py

```diff
@@ -5,20 +6,29 @@ class Loss(object):
     def __init__(self, model, dataset):
         self.model = model
         self.dataset = dataset
-        self.target = dataset.symb_targets
         self.consider_constant = []  # Part of the computational graph to be considered as a constant.
+        self.graph_updates = OrderedDict()
+        self._gradients = None
+
+    @property
+    def target(self):
+        return self.dataset.symb_targets
 
     def get_graph_output(self):
         output, updates = self.model.get_model_output(self.dataset.symb_inputs)
```

Comment:
Are you saying we should have `self.input` as well? I'm not even it makes sense semantically to have a `self.target` for a `Loss` object. Maybe we could make those properties private (e.g. `self._target`), what do you think?